### PR TITLE
Allow nodejs actions to run longer than 2 minutes

### LINKED ISSUE #1714: Action errors when invocation duration is longer than 2 minutes
## Steps to reproduce the issue:
When trying to recreate a user issue the other day I stumbled upon what seems to be an error with the OW backend.  When I created a NodeJS action that had a long execution duration (120000ms) and then created a trigger and rule for that action it produced an error once the action finished executing.  If I changed the duration of the action to be shorter (2000ms) it worked.  I created the action with a timeout of 300000ms (5min)

The error produced is `The action did not produce a valid response and exited unexpectedly.`

I used the following as my sample code. 
```
function main(params) {
  var time = 120000;
  return new Promise((resolve) => setTimeout(resolve, time));
}
```
1.  `wsk action create longAction ~/<pathToFile>.js -t 300000`
2.   I then went to the OW UI and created a Periodic trigger at 4 minute intervals and then created a rule with that trigger and the above action.
3.   After the 4 minutes the trigger fired, the rule fired, and the action was invoked.  
4.  After 2 minutes the action finished but it produced the error mentioned above. 
5.  I updated the action to finish after 2000ms and then ran `wsk action update longAction ~/<pathToFile>.js
6.  After 4 minutes, the trigger->rule->action fired.  The action completed successfully after the 2s.
7.  Updated the action again to have a 2 minute duration again and the action failed after it was invoked by the rule again. 

Here is the output of one of the failing invocations of my action
```
cordovaios:~/Dev/whiskStuff/ui$ wsk action invoke long
ok: invoked /_/long with id 51abb976f3934795b4a172ab76cdb032
cordovaios:~/Dev/whiskStuff/ui$ wsk activation get 51abb976f3934795b4a172ab76cdb032
ok: got activation 51abb976f3934795b4a172ab76cdb032
{
    "namespace": "eweiter@us.ibm.com_emptynew",
    "name": "long",
    "version": "0.0.6",
    "publish": false,
    "subject": "eweiter@us.ibm.com",
    "activationId": "51abb976f3934795b4a172ab76cdb032",
    "start": 1484256390062,
    "end": 1484256510167,
    "duration": 120105,
    "response": {
        "status": "action developer error",
        "statusCode": 0,
        "success": false,
        "result": {
            "error": "The action did not produce a valid response and exited unexpectedly."
        }
    },
    "logs": [
        "2017-01-12T21:26:30.125708738Z stdout: {}",
        "2017-01-12T21:26:30.125874727Z stdout: starting to wait"
    ],
    "annotations": [
        {
            "key": "limits",
            "value": {
                "logs": 10,
                "memory": 256,
                "timeout": 300000
            }
        },
        {
            "key": "path",
            "value": "eweiter@us.ibm.com_emptynew/long"
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by csantanapr**
Node js http server default timeout is 2 minutes, this PR changes the default timeout to not timeout, and the invoker will take care of the timeout.
Fixes #1714

### CODE REVIEW

**Comment by rabbah** [comment]
See the tests here: https://github.com/openwhisk/openwhisk/pull/1744/files
Feel free to cherry pick and then close my pr.

**Comment by csantanapr** [comment]
PG2 907 running...

File: core/nodejsActionBase/src/service.js [hunk 1]
```diff
@@ -63,7 +63,8 @@ function NodeActionService(config, logger) {
             var host = server.address().address;
             var port = server.address().port;
         });
-    }
+        server.timeout = 0;
```
Comment (rabbah) [inline core/nodejsActionBase/src/service.js]: add comment that this disable the auto disconnect.

File: tests/src/whisk/core/WskDeadlineLongTests.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (rabbah) [inline tests/src/whisk/core/WskDeadlineLongTests.scala]: delete this file?

File: tests/src/whisk/core/WskDeadlineLongTests.scala [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+   extends TestHelpers
+    with WskTestHelpers {
+
+    behavior of "Whisk deadline long running"
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk
+    val defaultAction = Some(TestUtils.getTestActionFilename("hello.js"))
+
+    it should s"invoke a nodejs action that returns a result by the deadline of 5 minutes" in withAssetCleaner(wskprops) {
```
Comment (rabbah) [inline tests/src/whisk/core/WskDeadlineLongTests.scala]: do we need this test?
it's testing that the action can run for 300 seconds and complete successfully.
so this is testing that the deadline is ok - which is already tested in WskBasicUsageTests for a lower latency.
to test that an action runs for the full 5 minutes, it's ok to just check that the duration of the action is at least 5 minutes as the other new test suite does (which would have failed before your patch above for node).

File: tests/src/whisk/core/WskDeadlineLongTests.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        (wp, assetHelper) =>
+            val name = "deadline"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("helloDeadline.js")), timeout = Some(300 seconds))
+            }
+
+            val run = wsk.action.invoke(name)
+            withActivation(wsk.activation, run, totalWait = 310 seconds) {
```
Comment (rabbah) [inline tests/src/whisk/core/WskDeadlineLongTests.scala]: the default poll frequency is too high for a test that's expected to last this long.

File: tests/src/whisk/core/limits/MaxActionDurationTests.scala [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+ */
+@RunWith(classOf[JUnitRunner])
+class MaxActionDurationTests
+    extends TestHelpers
+    with WskTestHelpers {
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk
+
+    Map("node" -> "helloDeadline.js", "python" -> "timedout.py").foreach {
```
Comment (rabbah) [inline tests/src/whisk/core/limits/MaxActionDurationTests.scala]: as noted earlier, these tests are sufficient.
if you add a jar for a java version, you can add it to the map to complete the runtime coverage.

File: tests/dat/actions/timedout.py [hunk 6]
```diff
@@ -0,0 +1,3 @@
+def main(dict):
+    while True:
+        0
```
Comment (rabbah) [inline tests/dat/actions/timedout.py]: in retrospect, i should have used a sleep so as not to churn out cpu load for no reason.

File: tests/src/whisk/core/WskDeadlineLongTests.scala (hunk 3 above)
Comment (csantanapr) [inline tests/src/whisk/core/WskDeadlineLongTests.scala]: yeah it was a mistake, I think it was as dealt with git this got in. Just removed

File: tests/src/whisk/core/limits/MaxActionDurationTests.scala (hunk 5 above)
Comment (csantanapr) [inline tests/src/whisk/core/limits/MaxActionDurationTests.scala]: I don't have a java action for this.

Comment (csantanapr) [inline tests/src/whisk/core/limits/MaxActionDurationTests.scala]: I added a jar for timedout

**Comment by csantanapr** [comment]
PG2 914 running...

**Comment by rabbah** [comment]
I have only one minor request: include the Java source for the corresponding jar. When you do, squash all the commit, no new PG needed to merge.

**Comment by csantanapr** [comment]
@rabbah java file added and squashed.

**Pull request merged**